rust/arrow/src/datatypes.rs

```diff
@@ -156,6 +157,107 @@ make_type!(UInt64Type, u64, DataType::UInt64, 64, 0u64);
 make_type!(Float32Type, f32, DataType::Float32, 32, 0.0f32);
 make_type!(Float64Type, f64, DataType::Float64, 64, 0.0f64);
 
+macro_rules! make_temporal_type {
+    ($name:ident, $native_ty:ty, $data_ty:expr, $bit_width:expr, $default_val:expr) => {
+        pub struct $name {}
+
+        impl ArrowPrimitiveType for $name {
+            type Native = $native_ty;
+
+            fn get_data_type() -> DataType {
+                $data_ty
+            }
+
+            fn get_bit_width() -> usize {
+                $bit_width
+            }
+
+            fn default_value() -> Self::Native {
+                $default_val
+            }
+        }
+    };
+}
+
+make_temporal_type!(
+    TimestampSecondType,
```

Comment:
Turns out I can't create a single `TimestampType` that can cater for all `TimeUnit` enum variants due to:

```rust
fn get_data_type() -> DataType {
    $data_ty
}
```

This is because when implementing the above trait member, I can't know at runtime what `TimeUnit` the array is implementing.
Is there a better/cleaner way of doing this?

Reply:
I'm thinking that similar to the C++ implementation, perhaps it's better to have the `TimeUnit` as a field to the `TimestampType`. For this, we'll need to change `DataType` so that date, time, and timestamp types won't carry unit.

Perhaps we can address this in a follow-up PR though.